helper/cli.c

```diff
@@ -0,0 +1,482 @@
+{
+	if (check_num_args(cli, argc, 1))
+		return CLI_ERROR;
+
+	odp_pool_t hdl = odp_pool_lookup(argv[0]);
+
+	if (hdl == ODP_POOL_INVALID) {
+		cli_error(cli, "%% Name not found.");
+		return CLI_ERROR;
+	}
+
+	odp_pool_print(hdl);
+
+	return CLI_OK;
+}
+
+static int cmd_call_odp_queue_print(struct cli_def *cli,
+				    const char *command ODP_UNUSED,
+				    char *argv[], int argc)
+{
+	if (check_num_args(cli, argc, 1))
+		return CLI_ERROR;
+
+	odp_queue_t hdl = odp_queue_lookup(argv[0]);
+
+	if (hdl == ODP_QUEUE_INVALID) {
+		cli_error(cli, "%% Name not found.");
+		return CLI_ERROR;
+	}
+
+	odp_queue_print(hdl);
+
+	return CLI_OK;
+}
+
+static int cmd_call_odp_shm_print(struct cli_def *cli,
+				  const char *command ODP_UNUSED, char *argv[],
+				  int argc)
+{
+	if (check_num_args(cli, argc, 1))
+		return CLI_ERROR;
+
+	odp_shm_t hdl = odp_shm_lookup(argv[0]);
+
+	if (hdl == ODP_SHM_INVALID) {
+		cli_error(cli, "%% Name not found.");
+		return CLI_ERROR;
+	}
+
+	odp_shm_print(hdl);
+
+	return CLI_OK;
+}
+
+static struct cli_def *create_cli(void)
+{
+	struct cli_command *c;
+	struct cli_def *cli;
+
+	cli = cli_init();
+	cli_set_banner(cli, NULL);
+	cli_set_hostname(cli, "ODP");
+
+	c = cli_register_command(cli, NULL, "show", NULL,
+				 PRIVILEGE_UNPRIVILEGED, MODE_EXEC,
+				 "Show information.");
+	cli_register_command(cli, c, "cpu", cmd_show_cpu,
+			     PRIVILEGE_UNPRIVILEGED, MODE_EXEC,
+			     "Show CPU information.");
+	cli_register_command(cli, c, "version", cmd_show_version,
+			     PRIVILEGE_UNPRIVILEGED, MODE_EXEC,
+			     "Show version information.");
+
+	c = cli_register_command(cli, NULL, "call", NULL,
+				 PRIVILEGE_UNPRIVILEGED, MODE_EXEC,
+				 "Call ODP API function.");
+	cli_register_command(cli, c, "odp_ipsec_print",
+			     cmd_call_odp_ipsec_print,
+			     PRIVILEGE_UNPRIVILEGED, MODE_EXEC, NULL);
+	cli_register_command(cli, c, "odp_pktio_print",
+			     cmd_call_odp_pktio_print,
+			     PRIVILEGE_UNPRIVILEGED, MODE_EXEC, "<name>");
+	cli_register_command(cli, c, "odp_pool_print",
+			     cmd_call_odp_pool_print,
+			     PRIVILEGE_UNPRIVILEGED, MODE_EXEC, "<name>");
+	cli_register_command(cli, c, "odp_queue_print",
+			     cmd_call_odp_queue_print,
+			     PRIVILEGE_UNPRIVILEGED, MODE_EXEC, "<name>");
+	cli_register_command(cli, c, "odp_shm_print_all",
+			     cmd_call_odp_shm_print_all,
+			     PRIVILEGE_UNPRIVILEGED, MODE_EXEC, NULL);
+	cli_register_command(cli, c, "odp_shm_print",
+			     cmd_call_odp_shm_print,
+			     PRIVILEGE_UNPRIVILEGED, MODE_EXEC, "<name>");
+	cli_register_command(cli, c, "odp_sys_config_print",
+			     cmd_call_odp_sys_config_print,
+			     PRIVILEGE_UNPRIVILEGED, MODE_EXEC, NULL);
+	cli_register_command(cli, c, "odp_sys_info_print",
+			     cmd_call_odp_sys_info_print,
+			     PRIVILEGE_UNPRIVILEGED, MODE_EXEC, NULL);
+
+	return cli;
+}
+
+static int cli_server(void *arg ODP_UNUSED)
+{
+	cli_shm_t *shm = (cli_shm_t *)odp_shm_addr(odp_shm_lookup(shm_name));
```

Comment:
The return value of `odp_shm_lookup()` should be checked first. if I remember correctly, Coverity will complain otherwise about unchecked return value.